### PR TITLE
Add extra tests for zero major version

### PR DESCRIPTION
**Pull request by m1kola**
# Description

Zero major version is reserved for initial development according to [SemVer spec](https://semver.org/#spec-item-4) and anything may change at any time.

As the result of the above update behaviour is different for it.

This is a follow up for https://github.com/operator-framework/operator-controller/pull/444#discussion_r1364836753

## Reviewer Checklist

- [ ] API Go Documentation
- [ ] Tests: Unit Tests (and E2E Tests, if appropriate)
- [ ] Comprehensive Commit Messages
- [ ] Links to related GitHub Issue(s)

### CODE REVIEW

File: internal/resolution/variablesources/installed_package_test.go [hunk 1]
```diff
@@ -118,33 +183,75 @@ func TestInstalledPackageVariableSource(t *testing.T) {
...
-		require.Len(t, bundles, 3)
-		assert.Equal(t, "test-package.v2.2.0", packageVariable.Bundles()[0].Name)
-		assert.Equal(t, "test-package.v2.1.0", packageVariable.Bundles()[1].Name)
-		assert.Equal(t, "test-package.v2.0.0", packageVariable.Bundles()[2].Name)
+			// ensure bundles are in version order (high to low)
+			bundles := packageVariable.Bundles()
+			require.Len(t, bundles, 3)
+			assert.Equal(t, "test-package.v2.2.0", packageVariable.Bundles()[0].Name)
+			assert.Equal(t, "test-package.v2.1.0", packageVariable.Bundles()[1].Name)
+			assert.Equal(t, "test-package.v2.0.0", packageVariable.Bundles()[2].Name)
```
Comment (m1kola) [inline internal/resolution/variablesources/installed_package_test.go:208]: This piece was shifted right into a sub test. Might be easier to review this PR ignoring whitespace changes (GH has a checkbox under the cog for this or usign `git diff -w`)

**Review by everettraven — APPROVED** [review]
LGTM
